src/ck-core/cklocation.C

```diff
@@ -2935,10 +2929,20 @@ int CkLocMgr::deliverMsg(CkArrayMessage* msg, CkArrayID mgr, CmiUInt8 id,
 #  else
     const CmiUInt8 lbObjId = id;
 #  endif
-    lbmgr->Send(
-        myLBHandle, lbObjId, UsrToEnv(msg)->getTotalsize(), cache->lastKnown(id), 1);
+    lbmgr->Send(myLBHandle, lbObjId, bytes, cache->lastKnown(id), 1);
   }
 #endif
+}
+
+/************************** LocMgr: MESSAGING *************************/
+/// Deliver message to this element, going via the scheduler if local
+/// @return 0 if object local, 1 if not
+int CkLocMgr::deliverMsg(CkArrayMessage* msg, CkArrayID mgr, CmiUInt8 id,
+                         const CkArrayIndex* idx, CkDeliver_t type, int opts)
+{
+  CkLocRec* rec = elementNrec(id);
+
+  recordSend(idx, id, UsrToEnv(msg)->getTotalsize(), opts);
```

Comment:
I'm pretty sure this will now record a send for the initial send, and every time the message is delivered. If you move the recordSend call to sendMsg, that should fix the issue. Although I'm not sure how it should interact with buffering.

Reply:
Should be fixed in 95cbdf7.